### ISSUE TITLE
Optionally allow lesson endpoints to be connected to the Net for controlled deployment 

### ISSUE THREAD

**Issue by olberger** [nre-learning/antidote#47]
The docs mention that : "For security reasons, network access outside the lesson namespace is disabled. All lessons should be totally self-contained and not rely on external resources to properly function when the lesson is being run."

I guess this is a reasonable expectation for a public platform like NRE Labs.

However, in order to make Antidote potentially deployed in more controlled environments (like a classroom/ university lab) it may be interesting to let learners use the Internet.

For instance, a Git practice lab could involve committing to an external repo like GitHub (well... one could also host a Gitlab instance inside the k8s cluster, I suppose ;).

It would be great to have the possibility to optionally deploy Antidote without such a restriction.

Hope that makes sense

**Comment by Mierdin** [comment]
Yeah this makes sense. And it's worth admitting this is much more of an NRE Labs restriction than an Antidote one.

We could just add an environment var to Syringe that lets you skip the addition of the NetworkPolicy that cuts off access out of the pod. So you have the option, but it's opt-in. So folks have to really mean it if they want to take on this risk themselves. I'm more in favor of this approach, since it continues to default to a culture of putting things in the image as opposed to relying on things to happen dynamically at runtime - while this restriction was created for NRE labs in the first place, I do see it having this nice side effect.

Anyways, WDYT?

**Comment by Mierdin** [comment]
@olberger Any thoughts? If the above looks okay, this is probably fairly easy to implement, and will make it into 0.4.0, at least in the mentioned form.
